src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2171,6 +2174,18 @@ public static void Select(IList? checkRead, IList? checkWrite, IList? checkError
             }
         }
 
+        public static void Select(IList? checkRead, IList? checkWrite, IList? checkError, TimeSpan timeout) => Select(checkRead, checkWrite, checkError, ToTimeoutMicroseconds(timeout));
+
+        private static int ToTimeoutMicroseconds(TimeSpan timeout)
+        {
+            long totalMicroseconds = (long)timeout.TotalMilliseconds * 1000;
```

Comment:
shouldn't this cast to long after the multiplication? as it is, you're losing any fractional milliseconds.

Reply:
It could also do `timeout.Ticks / TicksPerMicrosecond`. That seems more direct and more efficient. Arguably, there should be a `TimeSpan.TotalMicroseconds`.

Reply:
If I recall correctly `TotalMilliseconds` is simple enough (something like `Ticks / TicksPerMillisecond` to be optimized by the JIT.
@danmoseley What do you think?

@GSPP There is a api-suggestion for `TotalMicroseconds`: https://github.com/dotnet/runtime/issues/23799.
Since this API is apparently desired (since it has the `api-needs-work` label), we could assume that this will be implemented in the future we could add a link as a comment.

Reply:
@deeprobin it is best to verify not guess these things.

It seems that the division is smaller, for some reason. (@GSPP has more expertise here and can say whether it matters)
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBUALWbAEwEsAdgHMA3DRrEAzE1IMAwgwDeNBqqbTBGBgEEAFGz74YAZQAO2AQwy4AlEpVrHAGwjCrEDNicBZPmCgQuDCQAjy4DAC8DLouwja61uweXr5OTnxBIWEMAFQM5HSFNmLUDo6qfABm0RjJPn4Bma7ZADwMcOQMhITunvX+gcHN4QB8DJos3tgIAGpeAK4wNmXlytTlG+UYXBAA7gwCMPvaUELzRgIYAPLz15UAShZCMACiCGAwphh8rroC2EYIJUErZiitHABfcFqYgAdmimhstT6vgGTVCuBKkJW0NUUnGlwYACF9IYTOZLNY7GtNgxYkJeikGoMsuEojFXEJ4okDGAANbhAD0+SKJVx42qCTqqMaQwxDDaHS6PWRTLRcuyYwmU1mCyW4pptM22wC+0Ox1O5xglxud0ewle70+31+/0BwKpYPWmyh3o2cIRlyR0uZ6LCWLUvohQA

Reply:
I'm basing my argument mostly on cleanliness of the design. I find it, intuitively, a bit ugly to double-convert (and have floating-point computation in the mix). It all might work out concretely, but I'd find the code "saner" if it just didn't go down that path.

In my opinion, it also would be easier to review and verify for correctness. Less moving parts.

Reply:
> I'm surprised that we see a `vdivsd` instruction at all. Aren't constant divisors being optimized in the JIT? Or is that just for some hard-coded, common values?

Citation @saucecontrol - C# Community Discord:
> you can multiply by reciprocal, but that's not guaranteed to be the same result

Therefore, this division is probably not optimized as we would like it to be.

Reply:
Of course, this is floating point...